docs/concepts/pdp/factdb.mdx

```diff
@@ -0,0 +1,230 @@
+reducing the latency and improving the performance of the PDP.
+
+:::info EAP
+This feature is available under Early Access Program.
+Please contact with us on [Slack](https://io.permit.io/slack) for more information.
+:::
+
+:::warning
+This version is available on **PDP v0.7.0** and above.
+Please make sure to upgrade **all PDPs in the environment** to the latest version to use FactDB.
+:::
+
+## Enabling FactDB
+
+The FactDB can be enabled per Permit Environment, allowing PDPs in that environment to use the FactDB.
+Optionally, you can have both legacy PDPs and FactDB enabled PDPs in the same environment.
+
+To enable FactDB on a Permit Environment, you'll need an [API Key](/manage-your-account/workspace-settings#api-keys),
+then follow the steps below:
+
+#### 1. Allow FactDB on Permit.io Environment Settings
+
+Update your environment settings to enable FactDB.
+```bash
+curl -X PATCH 'https://api.permit.io/v2/projects/{proj_id}/envs/{env_id}' \
+  -H 'Content-Type: application/json' -H 'Accept: application/json' \
+  -H 'Authorization : Bearer {api_key}' \
+  --data-raw '{
+      "settings": {
+            "factdb_enabled":true
+      }
+    }'
+```
+:::info
+If you have existing settings in your environment, you may want to include them in the PATCH request.
+Check it using the following request:
+```bash
+curl -X GET 'https://api.permit.io/v2/projects/{proj_id}/envs/{env_id}' \
+  -H 'Accept: application/json' -H 'Authorization : Bearer {api_key}'
+```
+:::
+
+
+#### 2. Enable FactDB on your PDPs
+
+Set the `PDP_FACTDB_ENABLED=true` environment variable to enable FactDB on your all PDPs connected to the environment.
+
+```bash
+docker run -it -p 7766:7000 \
+  --name pdp-online \
+  --env PDP_API_KEY=<YOUR_API_KEY> \
+  --env PDP_FACTDB_ENABLED=true \
+  permitio/pdp-v2:latest
+```
+
+:::info
+FactDB is currently an opt-in feature, requiring you to explicitly enable it on all PDPs in the environment.
+This is to allow environments to run both with FactDB and without it.
+:::
+
+### Running both FactDB and Legacy PDPs in the same environment
+
+For simplifying migration plan, it is possible to run both FactDB enabled PDPs and Legacy PDPs in the same environment.
+To do this, you need to enable the following setting on the environment.
+
+```bash
+curl -X PATCH 'https://api.permit.io/v2/projects/{proj_id}/envs/{env_id}' \
+  -H 'Content-Type: application/json' -H 'Accept: application/json' \
+  -H 'Authorization : Bearer {api_key}' \
+  --data-raw '{
+      "settings": {
+            "factdb_enabled":true,
+            "allow_legacy_data_store":true
+      }
+    }'
+```
+
+With this setting enabled, you can run both FactDB PDPs with `PDP_FACTDB_ENABLED=true` and Legacy PDPs in the same environment.
+
+:::info
+We recommend you to migrate all your PDPs in the environment to FactDB.
+Please contact us on [Slack](https://io.permit.io/slack) for more information.
+:::
+
+
+### PDP Offline Mode with FactDB
```

Comment:
Is it unique to FactsDB?
If not let's move it to offline mode section and add that factsDB is supporting this feature as well (if you want)

Reply:
No, but @orweis commented that we should explain how it works with Offline Mode, and I agree that we should emphasis on it here as people are likely going to question that